ui/lib/config-ui/addon/routes/messages/index.js

```diff
@@ -28,36 +30,49 @@ export default class MessagesRoute extends Route {
     },
   };
 
-  model(params) {
+  async model(params) {
     const { authenticated, page, pageFilter, status, type } = params;
-    const filter = pageFilter
-      ? (dataset) => dataset.filter((item) => item?.title.toLowerCase().includes(pageFilter.toLowerCase()))
-      : null;
-    let active;
+    const active = {
+      active: true,
+      inactive: false,
+    }[status];
 
-    if (status === 'active') active = true;
-    if (status === 'inactive') active = false;
-
-    const messages = this.pagination
-      .lazyPaginatedQuery('config-ui/message', {
-        authenticated,
-        pageFilter: filter,
+    try {
+      const { keyInfo, keys } = await this.api.sys.uiConfigListCustomMessages(
+        true,
         active,
-        type,
-        responsePath: 'data.keys',
-        page: page || 1,
-        size: 10,
-      })
-      .catch((e) => {
-        if (e.httpStatus === 404) {
-          return [];
-        }
-        throw e;
+        authenticated,
+        type
+      );
+      // ids are in the keys array and can be mapped to the object in keyInfo
+      // map and set id property on keyInfo object
+      const data = keys.map((id) => {
+        const { startTime, endTime, ...message } = keyInfo[id];
+        // dates returned from list endpoint are strings -- convert to date
+        return {
+          id,
+          ...message,
+          startTime: startTime ? new Date(startTime) : startTime,
+          endTime: endTime ? new Date(endTime) : endTime,
+        };
       });
-    return hash({
-      params,
-      messages,
-    });
+      const messages = paginate(data, {
+        page,
+        pageSize: 2,
+        filter: pageFilter,
```

Comment:
It looks like this was passing a function before, and now it's passing the filter param. Does `paginate()` handle filtering now?

Reply:
Ah yes, I see the "filterKey" below

Reply:
Yes I basically extracted the pagination functionality from `lazyPaginatedQuery` out to the util and preserved the ability to filter an array of objects by a given property.